R/collector.R

```diff
@@ -52,8 +57,16 @@ set_collector <- function(funs = NULL, pkg = NULL, path = "collector") {
     # Replace bodies a call to collect_and_run(), keep original call as attr
     for (nm in names(funs)) {
       val <- ns[[nm]]
+      if (length(formals(val)) > 0 && names(formals(val))[[1]] != "...") {
```

Comment:
Does this force every first argument ? I can think of a few functions with a NSE 1st arg, like quote, substitute, ~, expr, quo, unquo, inject.
This is also not "true" collection because we lose the logic on how the data was built.
I get that it is useful but do we need an argument to switch this on ?

Reply:
Yeah, the results aren't great here either. We need to find a better way to deal with nested calls. The next step would be to create a reprex to better understand the problem.

Reply:
By nested call do you mean that with something like : 

```
df |>
  mutate(...) |>
  select(...)
```

We'd like to collect both calls ? if so `globals$collecting` is indeed preventing it, the older behavior wasn't.
Since we don't collect when a function is called from its namespace (or a child of its namespace) the `globals$collecting` is really doing just that as far as I understand.

I can propose a `force` argument, a named list where names are functions and values a vector of argument names.
This list could be pre-computed from static analysis, and then fixed manually.